Add uxr_prepare_output_stream_fragmented API (#52)
eProsima/Micro-XRCE-DDS-docs: docs/client.rst

------

.. code-block:: c
    bool uxr_prepare_output_stream_fragmented(uxrSession* session, uxrStreamId stream_id, uxrObjectId datawriter_id,

>                                   struct ucdrBuffer* ub, size_t topic_size, uxrOnBuffersFull flush_callback);

Reviewer: struct *ucdr buffer ub? ub_topic? mb? mb_topic?

Author: the name does not mind... just the type. In any case, in the actual library they have different names.

Reviewer: Yes, of course. I mean, just to be consistent w/ the description given below.

Author: Done

> Requests a writing into a specific output stream.

Reviewer: ```suggestion
Requests to allocate an output stream of ``topic_size`` bytes for a write operation.
```

> This function will initialize an ``ucdrBuffer`` struct where a topic of ``topic_size`` size must be serialized.

Reviewer: ```suggestion
This function will initialize an ``ucdrBuffer`` struct where a topic of ``topic_size`` size will be serialized.
```

> Whether the necessary gap for writing a ``topic_size`` bytes into the reliable stream, the returned value is ``true``, otherwise ``false``.

Reviewer: ```suggestion
If there is sufficient space for writing ``topic_size`` bytes into the reliable stream, the returned value is ``true``, otherwise it is ``false``.
```

> The topic will be sent in the next ``run_session`` function. If buffer is full at some moment during the serialization process, ``flush_callback`` will be call and user will be in charge of running a session for flushing the reliable stream.

Reviewer: ```suggestion
The topic will be sent in the following ``run_session`` function. If during the serialization process the buffer gets overfilled, the ``flush_callback`` function will be called and the user will be in charge of running a session for flushing the stream.
```

NOTE: This approach is not valid with best-effort streams.

:session: Session structure previously initialized.
:stream_id: The output stream ID where the message will be written.
:datawriter_id: The `datawriter` ID that will write the topic to the DDS World.

> :mb_topic: An ``ucdrBuffer`` struct used to serialize the topic.

Reviewer: choose a consistent name for the :mb_topic: (see above).

Author: Done

>            This struct points to a requested gap into the stream.

Reviewer: ```suggestion
           This struct points to a requested memory slot in the stream.
```

> :topic_size: The bytes that will be reserved in the stream.

Reviewer: ```suggestion
:topic_size: The slot, in bytes, that will be reserved in the stream.
```
